### ISSUE TITLE
Bug when compiling in ubuntu (Raspberry Pi)

### ISSUE THREAD

**Issue by Jesus82** [roc-streaming/roc-toolkit#439]
I'm trying to compile in a Raspberry Pi 4 with Ubuntu 20.10 freshly installed.

When doing `scons -Q --enable-pulseaudio-modules --build-3rdparty=openfec,pulseaudio`, I get `can't make 'json-0.11-20130402', see '3rdparty/aarch64-pc-linux-gnu/gcc-10.2.0-release/build/json-0.11-20130402/build.log' for details`
The logs:
```
checking build system type... ./config.guess: unable to guess system type

This script, last modified 2009-12-30, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
and
  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

If the version you run (./config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.

config.guess timestamp = 2009-12-30

uname -m = aarch64
uname -r = 5.8.0-1006-raspi
uname -s = Linux
uname -v = #9-Ubuntu SMP PREEMPT Fri Oct 16 12:55:30 UTC 2020

/usr/bin/uname -p = aarch64
/bin/uname -X     = 

hostinfo               = 
/bin/universe          = 
/usr/bin/arch -k       = 
/bin/arch              = aarch64
/usr/bin/oslevel       = 
/usr/convex/getsysinfo = 

UNAME_MACHINE = aarch64
UNAME_RELEASE = 5.8.0-1006-raspi
UNAME_SYSTEM  = Linux
UNAME_VERSION = #9-Ubuntu SMP PREEMPT Fri Oct 16 12:55:30 UTC 2020
configure: error: cannot guess build type; you must specify one
```
Is there any manual way to tell the script which OS I'm running? 

Thanks in advance!

**Comment by gavv** [comment]
Hi! Sorry for delay.

`config.guess` should return a string like `x86_64-pc-linux-gnu`. You can define it manually using `--build` option (e.g. `--build=x86_64-pc-linux-gnu`). If you do this, config.guess is not used by Roc.

To find out what string you need, you can run a fresh copy of config.guess script manually (e.g. download it from GNU web site), or try the output from this command: `gcc -v -E - </dev/null |& grep Target`.

You can also get a fresh copy of config.guess and configure Roc to use it by setting environment variable `CONFIG_GUESS` to the path of the script.

I hope one of these will solve the problem. Anyway, I'll try to reproduce it by myself and maybe implement a fix so that it can work out of the box.

**Comment by Jesus82** [comment]
When doing `scons -Q --enable-pulseaudio-modules --build=aarch64-linux-gnu `i got `error: openfec not found (see 'config.log' for details)`

So I downloaded a fresh `config.guess `from [http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD](http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD), and replaced the original one in three different places, that worked. Thanks for the input!

**Comment by gavv** [comment]
Interesting, it's reproducing only on Ubuntu 20.10 on 64-bit ARM, but not on 32-bit ARM or x86_64.

I finally was able to reproduce and fix it: ef5374bc92f1d9e73185b7db2e78e1ec62dc9d38. It should now build out of the box.

Thanks for report! Feel free to reopen this if you'll have any further problems.
